base/base.bzl

```diff
@@ -3,52 +3,74 @@ load("@io_bazel_rules_docker//container:container.bzl", "container_image")
 load("@io_bazel_rules_docker//contrib:test.bzl", "container_test")
 load(":distro.bzl", "DISTRO_PACKAGES", "DISTRO_REPOSITORY")
 load("//cacerts:cacerts.bzl", "cacerts")
+load("//:checksums.bzl", "ARCHITECTURES")
 
 NONROOT = 65532
 
 # Replicate everything for debian9 and debian10
 def distro_components(distro_suffix):
-    cacerts(
+    for arch in ARCHITECTURES:
+        cacerts(
+            name = "cacerts_" + arch + distro_suffix,
+            deb = DISTRO_PACKAGES[arch][distro_suffix]["ca-certificates"],
+        )
+
+        for (user, uid, workdir) in [("root", 0, "/"), ("nonroot", NONROOT, "/home/nonroot")]:
+            container_image(
+                name = "static_" + user + "_" + arch + distro_suffix,
+                debs = [
+                    DISTRO_PACKAGES[arch][distro_suffix]["base-files"],
+                    DISTRO_PACKAGES[arch][distro_suffix]["netbase"],
+                    DISTRO_PACKAGES[arch][distro_suffix]["tzdata"],
+                ],
+                architecture = arch,
+                env = {
+                    "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                    # allows openssl to find the certificates by default
+                    # TODO: We should run update-ca-certifaces, but that requires "openssl rehash"
+                    # which would probably need to be run inside the container
+                    "SSL_CERT_FILE": "/etc/ssl/certs/ca-certificates.crt",
+                },
+                tars = [
+                    ":passwd",
+                    ":group_tar",
+
+                    # Create /tmp, too many things assume it exists.
+                    # tmp.tar has a /tmp with the correct permissions 01777
+                    # A tar is needed because at the moment there is no way to create a
+                    # directory with specific permissions.
+                    ":tmp.tar",
+                    ":nsswitch.tar",
+                    DISTRO_REPOSITORY[arch][distro_suffix] + "//file:os_release.tar",
+                    ":cacerts_" + arch + distro_suffix + ".tar",
+                ],
+                user = "%d" % uid,
+                workdir = workdir,
+            )
+
+    for (user, uid, workdir) in [("root", 0, "/"), ("nonroot", NONROOT, "/home/nonroot")]:
+        native.alias(
+            name = "static_" + user + distro_suffix,
+            actual = ":static_" + user + "_amd64" + distro_suffix,
+        )
+
+    native.alias(
```

Comment:
would be good to add a comment on things like this mentioning that there's nothing architecture-specific in the deb

Reply:
That's not why this is here.  I'm following the pattern established in `BUILD`, which seems to be so that folks relying on the old target names aren't broken.

That said, I'm kinda inclined to rip them all out as part of this refactor, but without understanding the history better didn't want to jump the gun.

Reply:
fair enough